conda/recipes/ast_canopy/meta.yaml

```
@@ -20,29 +20,23 @@ requirements:
   build:
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
-    - {{ compiler('cuda') }}
     - cmake >=3.28
-    - pybind11
-    - clangdev >=18
     - ninja
-    - sysroot_{{ target_platform }} 2.17
+    - {{ stdlib("c") }}
   host:
-    - {{ compiler('c') }}
-    - {{ compiler('cxx') }}
-    - cuda-version {{ cuda_compiler_version }}
     - python
+    - pybind11
     - pip
     - scikit-build-core
+    - clangdev >=18
     - llvmdev >=14
     - llvm >=14
   run:
-    - {{ compiler('c') }}
-    - {{ compiler('cxx') }}
+    - cuda-version >=12.3
+    - cuda-nvcc >=12.3
+    - gcc >=10
```

Let's try this suggestion. The upper limit on CUDA 12.5 is currently `<13`, and it looks like conda is preferring to fall back to the (incorrectly pinned) `nvidia` channel packages from CUDA 12.4 with GCC 13 rather than use CUDA 12.5 with GCC 12.

This is "redundant" but it eliminates an unwanted result from the conda solution space, which should help.
```suggestion
    - gcc >=10,<13
```